dhall/src/Dhall/Core.hs

```diff
@@ -346,6 +346,9 @@ instance Pretty Var where
     pretty = Pretty.unAnnotate . prettyVar
 
 -- | Syntax tree for expressions
+
+-- NB: If you add a constructor, please also update the Arbitrary instance in
```

Comment:
Double-check that this comment doesn't get included in the generated haddocks

If it does, then you could move it underneath the `Expr` type (since that's where people are most likely to insert new constructors anyway)

Reply:
It's not included in the haddocks.

Reply:
I've moved the note anyway.